codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -773,7 +764,8 @@ private void verifyNotProcessingHeaders() throws Http2Exception {
 
     private void verifyPayloadLength(int payloadLength) throws Http2Exception {
         if (payloadLength > maxFrameSize) {
-            throw connectionError(PROTOCOL_ERROR, "Total payload length %d exceeds max frame length.", payloadLength);
+            throw connectionError(FRAME_SIZE_ERROR, "Total payload length %d exceeds max frame length %d.",
```

Comment:
This changes semantics as the error reason is changed... Do you say using `PROTOCOL_ERROR` is a bug ? If so we should also add a unit test test. @ejona86 thoughts ?

Reply:
Since the processHeaderState() check always ran before the `verify*()` checks, then changing this preserves the existing behavior. The previous PROTOCOL_ERROR path was not possible to execute. RFC7540§4.2 requires FRAME_SIZE_ERROR.

I don't really understand why processHeaderState() is now calling verifyPayloadLength() though; verifyPayloadLength() is only used once, so why not inline it which actually is a smaller change?

Reply:
Hi @ejona86 , @normanmaurer, thanks for the comments. I am not quite understand what you guys mean by inline it. Could you please give more details here? Is it like keeping the following if statement and removed the duplicate check payload length in `verify*()`? Thanks!

```
  if (payloadLength > maxFrameSize) {
      throw connectionError(FRAME_SIZE_ERROR, "Frame length: %d exceeds maximum: %d", payloadLength,
                            maxFrameSize);
  }
```

Reply:
Instead of deleting the payloadLength check and calling verifyPayloadLength(), I'd have expected you to just delete verifyPayloadLength() and the lines that call it; there was no need to modify processHeaderState(). By "inlining" I mean copy the contents of a method to the location that method is used, instead of calling the method.

It wasn't that big of a deal, but I think it would have made the diff more obvious and makes reading the code easier.

Reply:
I understand, thank you so much!